src/main/java/org/tron/core/capsule/TransactionCapsule.java

```diff
@@ -360,6 +362,25 @@ public void sign(byte[] privateKey) {
     }
   }
 
+  // todo mv this static function to capsule util
+  public static long getCpuLimitInTrx(Transaction.Contract contract) {
+    int cpuForTrx;
+    try {
+      Any contractParameter = contract.getParameter();
+      switch (contract.getType()) {
+        case TriggerSmartContract:
+          return contractParameter.unpack(TriggerSmartContract.class).getCpuLimitInTrx();
+        case CreateSmartContract:
+          return contractParameter.unpack(CreateSmartContract.class).getCpuLimitInTrx();
+        default:
+          return 0;
+      }
+    } catch (Exception ex) {
+      ex.printStackTrace();

```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1188&issues=AWTg8qUtOVxoJwmFDyWU&open=AWTg8qUtOVxoJwmFDyWU)

[ISSUE_KEY]: AWTg8qUtOVxoJwmFDyWU